src/query_sink.rs

```diff
@@ -65,14 +66,44 @@ impl AsyncQueryResultStreamImpl {
 }
 
 /// A stream of WMI query results.
+///
+/// When dropped, the stream is properly cancelled and the resources freed.
+#[derive(Clone)]
```

Comment:
Should `Clone` be implemented if you cancel on drop?

Reply:
Ah yes this is a mistake, I probably wanted to keep the Debug derive and not the Clone one.
But actually, i'll just remove the Debug derive as well, it allows removing the custom Debug impl for QuerySink, and since other public types are missing Debug impls, this is more consistent.